crypto/keys/src/ristretto.rs

```diff
@@ -105,7 +105,18 @@ impl PrivateKey for RistrettoPrivate {
 
 impl FromRandom for RistrettoPrivate {
     fn from_random<R: CryptoRng + RngCore>(csprng: &mut R) -> RistrettoPrivate {
-        Self(Scalar::random(csprng))
+        let mini_secret = MiniSecretKey::generate_with(csprng);
+        let secret = mini_secret
+            .expand_to_keypair(ExpansionMode::Uniform)
+            .secret
+            .clone();
+        // The scalar is the first 32 bytes in canonical representation
+        let mut scalar_bytes: [u8; 32] = [0u8; 32];
+        scalar_bytes[0..32].copy_from_slice(&secret.to_bytes()[0..32]);
+        let scalar = Scalar::from_canonical_bytes(scalar_bytes)
+            .ok_or(KeyError::InvalidPrivateKey)
+            .expect("Could not get scalar from RistrettoPrivate");
+        Self(scalar)
```

Comment:
How does the result of this implementation differ from `Self(Scalar::random(csprng))`? How is it better?

I tried to walk through this, and it looks like it generates a MiniSecreyKey containing a random [u8; 32], then "expands" it by creating a Merlin transcript containing the MiniSecretKey, then extracting another [u8; 32] from the transcript (i.e. the hash of the transcript), which becomes the final scalar. 

In total, that seems like we've basically replaces Scalar::random, which is implemented as:

```
let mut scalar_bytes = [0u8; 64];
        rng.fill_bytes(&mut scalar_bytes);
        Scalar::from_bytes_mod_order_wide(&scalar_bytes)
```
with something that ultimately calls Scalar::from_bytes_mod_order_wide on the 64-byte result of hashing a 32-byte random value (i.e. the MiniSecretKey).

Reply:
afaict the scalar part is not different from `Scalar::random(csprng)`,

the only thing it is doing is creating the Scalar and the nonce in a way that has a relationship similar to that defined in ed25519 spec (but they have a variation called "uniform"). the Scalar we end up with has exactly the same distribution as `Scalar::random(csprng)`. the nonce is only used when creating signatures.